### PR TITLE
Features/156 random addon

### PR DESCRIPTION
**Pull request by crankycoder**
Fix for #156 

This adds the ability to run an experimental branch of TAAR which randomizes the list of addons for a user.

By default, the experimental branch will not be executed, but you can override this with an environment variable in  `TAAR_EXPERIMENT_PROB`.

The addons are selected using `numpy.random.choice` with no  replacement.

The list of GUID weights is normalized to provide a `p` probabilities to the choice function.

For the sake of whitelist coverage - when the experimental branch is selected, all whitelisted addons selectable by the `choice` function, however the list is cut off using the usual `length` argument passed into the recommendation engine.

### CODE REVIEW

**Comment by mlopatka** [comment]
@Dexterp37 We cold use an objective look at this small patch. Expected to provide two final shuffling of recommended addons for later use in-service A/B testing.

File: tests/test_randomizer.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def test_reorder_guids():
+    guid_weight_tuples = [
+        ("guid1", 0.1),
+        ("guid2", 0.2),
+        ("guid3", 0.3),
+        ("guid4", 0.4),
+    ]
+    actual = reorder_guids(guid_weight_tuples)
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply tests/test_randomizer.py]: Thanks - I've added multiple runs and tuned the weights to make sure the results come back in a meaningful way for humans in 2016c08

File: tests/test_randomizer.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    print(actual)
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply tests/test_randomizer.py]: doh!  I've added a proper assertion here now in 2016c08

File: taar/recommenders/randomizer.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    xp_prob should be a probability between 0.0 and 1.0 which is the
+    chance that the experimental branch is selected.
+    """
+    hex_client = ''.join([c for c in client_id if c.lower() in 'abcdef0123456789'])
+    int_client = int(hex_client, 16)
+    return int((int_client % 100) <= (xp_prob * 100))
+
+
+def reorder_guids(guid_weight_tuples, size=None):
+    weight_list = [weight for (guid, weight) in guid_weight_tuples]
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply taar/recommenders/randomizer.py]: docstring added referring to the numpy.random.choice documentation

File: taar/recommenders/randomizer.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def in_experiment(client_id, xp_prob=0.5):
+    """
+    Return whether or not this client_id is in the experiment.
+
+    xp_prob should be a probability between 0.0 and 1.0 which is the
+    chance that the experimental branch is selected.
+    """
+    hex_client = ''.join([c for c in client_id if c.lower() in 'abcdef0123456789'])
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply taar/recommenders/randomizer.py]: Added comment in 15cb1d1

File: taar/recommenders/randomizer.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+"""
```
[the remark this answers is not in the document]
Comment (crankycoder) [reply taar/recommenders/randomizer.py]: added in c4f24e1

**Pull request merged**